### PR TITLE
fix(Core/WorldSocket): error C2102

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
- for c++20 build

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Winfidonarleyan** [comment]
Need test this

File: src/server/game/Server/WorldSocketMgr.cpp [hunk 1]
```diff
@@ -62,7 +62,7 @@ bool WorldSocketMgr::StartWorldNetwork(Acore::Asio::IoContext& ioContext, std::s
     if (!BaseSocketMgr::StartNetwork(ioContext, bindIp, port, threadCount))
         return false;
 
-    _acceptor->AsyncAcceptWithCallback<&OnSocketAccept>();
+    _acceptor->AsyncAcceptWithCallback<OnSocketAccept>();
```
Comment (FrancescoBorzi) [inline src/server/game/Server/WorldSocketMgr.cpp:65]: not sure about this tbh

**Review by FrancescoBorzi — CHANGES_REQUESTED** [review]

File: src/server/game/Server/WorldSocketMgr.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/game/Server/WorldSocketMgr.cpp:65]: So what do you suggest to do instead?

Comment (FrancescoBorzi) [reply src/server/game/Server/WorldSocketMgr.cpp:65]: @Kitzunu no idea.. unfortunately, I didn't have the opportunity to dig into this

**Comment by Kitzunu** [comment]
gonna close, I dont think this is the right approach

**Pull request closed without merging**